src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -66,617 +58,585 @@ const defaultProps = {
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    useEffect(() => {
+        // Screen coming back into focus, for example
+        // when doing Cmd+Shift+K, then Cmd+K, then Cmd+Shift+K.
+        // Only applies to platforms that support keyboard shortcuts
+        if ([CONST.PLATFORM.DESKTOP, CONST.PLATFORM.WEB].includes(getPlatform()) && isFocused && props.autoFocus && textInputRef.current) {
+            setTimeout(() => {
+                textInputRef.current.focus();
+            }, CONST.ANIMATED_TRANSITION);
         }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isFocused, props.autoFocus]);
 
-        const itemIndex = option.index;
-        const sectionIndex = option.sectionIndex;
+    useEffect(() => {
+        const newSections = sliceSections();
+        const newOptions = flattenSections();
 
-        if (!lodashGet(this.state.sections, `[${sectionIndex}].data[${itemIndex}]`, null)) {
+        if (prevLocale.current !== props.preferredLocale) {
+            prevLocale.current = props.preferredLocale;
+            setAllOptions(newOptions);
+            setSections(newSections);
             return;
         }
 
-        // Note: react-native's SectionList automatically strips out any empty sections.
-        // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
-        // Otherwise, it will cause an index-out-of-bounds error and crash the app.
-        let adjustedSectionIndex = sectionIndex;
-        for (let i = 0; i < sectionIndex; i++) {
-            if (_.isEmpty(lodashGet(this.state.sections, `[${i}].data`))) {
-                adjustedSectionIndex--;
-            }
+        const newFocusedIndex = props.selectedOptions.length;
+
+        setSections(newSections);
+        setAllOptions(newOptions);
+        setFocusedIndex(_.isNumber(props.focusedIndex) ? props.focusedIndex : newFocusedIndex);
+        // we want to run this effect only when the sections change
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.sections]);
+
+    useEffect(() => {
+        // If we just toggled an option on a multi-selection page or cleared the search input, scroll to top
+        if (props.selectedOptions.length !== prevSelectedOptions.current.length || (!!prevValue.current && !value)) {
+            prevSelectedOptions.current = props.selectedOptions;
+            prevValue.current = value;
+            scrollToIndex(0);
+            return;
         }
 
-        this.list.scrollToLocation({sectionIndex: adjustedSectionIndex, itemIndex, animated});
-    }
+        // Otherwise, scroll to the focused index (as long as it's in range)
+        if (allOptions.length <= focusedIndex) {
+            return;
+        }
+        scrollToIndex(props.focusedIndex);
```

Comment:
```suggestion
        scrollToIndex(focusedIndex);
```
Maybe this is not correct